src/base/inifile.cc

```diff
@@ -294,32 +285,29 @@ IniFile::Section::printUnreferenced(const std::string &sectionName)
 void
 IniFile::getSectionNames(std::vector<std::string> &list) const
 {
-    for (SectionTable::const_iterator i = table.begin();
-         i != table.end(); ++i)
-    {
-        list.push_back((*i).first);
+    for (auto& entry: table) {
+        auto& sectionName = entry.first;
+        list.push_back(sectionName);
     }
 }
 
 bool
-IniFile::printUnreferenced()
+IniFile::printUnreferenced() const
 {
     bool unref = false;
 
-    for (SectionTable::iterator i = table.begin();
-         i != table.end(); ++i) {
-        const std::string &sectionName = i->first;
-        Section *section = i->second;
+    for (auto& entry: table) {
+        auto& [sectionName, section] = entry;
```

Comment:
Do we know when GCC/clang started supporting this syntax? I want to make sure that all of our supported compilers support this.